launcher/common/xml_manager.py

```diff
@@ -0,0 +1,310 @@
+# ------------------------------------------------------------------------------
```

Comment:
I commented below on the naming of the XML manager classes that inherit from this one.

Otherwise very nice work. You made a huge effort to do this properly and extensible.